lib/track.js

```diff
@@ -0,0 +1,171 @@
+const Request = require('./request');
+const { trackRoot, apiRoot } = require('./common');
+const { isEmpty } = require('./utils');
+
+const BROADCASTS_ALLOWED_RECIPIENT_FIELDS = {
+  ids: ['ids', 'id_ignore_missing'],
+  emails: ['emails', 'email_ignore_missing', 'email_add_duplicates'],
+  per_user_data: ['per_user_data'],
```

Comment:
Just a reminder to merge in #32 before this one so they don't have to rebase their changes